spec/nine_one_one_spec.rb

```diff
@@ -171,33 +98,109 @@
         end.to raise_error(NineOneOne::ConfigurationError)
       end
     end
+
+    context 'multi config' do
+      it 'allows to have separate configs for different purposes' do
+        NineOneOne.configure(:urgent) do |config|
+          config.slack_enabled = false
+        end
+
+        NineOneOne.configure(:non_urgent) do |config|
+          config.slack_webhook_url = 'webhook'
+          config.slack_channel = 'channel'
+          config.slack_enabled = true
+        end
+
+        expect(NineOneOne.configs[:default].slack_enabled).to eq(false)
+        expect(NineOneOne.configs[:default].slack_enabled).to eq(false)
+        expect(NineOneOne.configs[:non_urgent].slack_enabled).to eq(true)
+      end
+
+      it 'raises error when non-existent configuration is requested' do
+        expect { NineOneOne.use(:foo) }.to raise_error(NineOneOne::NotConfiguredError)
+      end
+    end
+  end
+
+  describe '.use' do
+    context 'default config' do
+      it 'returns preconfigured notifier' do
+        NineOneOne.configure do |config|
+          config.send_pagers = true
+          config.pager_duty_integration_key = 'integration key'
+          config.slack_enabled = true
+          config.slack_channel = 'channel'
+          config.slack_webhook_url = 'http://example.com'
+        end
+
+        notifier = NineOneOne.use(:default)
```

Comment:
looks like something that could be defined as subject + 'configure' block could be relocated to the before block